packages/client/hmi-client/src/services/models/funman-service.ts

```diff
@@ -105,25 +106,19 @@ export const processFunman = (result: any) => {
 						return obj;
 					}, {});
 
-				const timesteps = [
-					...new Set(
-						Object.keys(filteredVals).map((key) => {
-							const splitKey = key.split('_');
-							return +splitKey[splitKey.length - 1]; // timestep
-						})
-					)
-				].sort((a, b) => a - b);
-
 				timesteps.forEach((t) => {
 					const traj: any = {
 						boxId,
 						pointId: point.id,
 						timestep: t
 					};
 					states.forEach((s) => {
-						traj[s] = filteredVals[`${s}_${t}`];
+						if (filteredVals[`${s}_${t}`] !== undefined) {
```

Comment:
`!!` will check for not being `undefined` or `null`. Unless you really only want to check if it's not empty.
Otherwise use Lodash `!isEmtpy()`

```suggestion
						if (!!filteredVals[`${s}_${t}`]) {
```
```suggestion
						if (!isEmpty(filteredVals[`${s}_${t}`])) {
```